src/auth/src/credentials/api_key_credential.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::credentials::dynamic::CredentialTrait;
+use crate::credentials::{Credential, Result, QUOTA_PROJECT_KEY};
+use crate::errors::CredentialError;
+use crate::token::{Token, TokenProvider};
+use http::header::{HeaderName, HeaderValue};
+use std::sync::Arc;
+
+const API_KEY_HEADER_KEY: &str = "x-goog-api-key";
+
+/// Configuration options for an API key credential.
+#[derive(Default)]
+pub struct ApiKeyOptions {
+    quota_project: Option<String>,
+}
+
+impl ApiKeyOptions {
+    /// Set the [quota project].
+    ///
+    /// If unset, the project associated with the API key will be used as the
+    /// quota project.
+    ///
+    /// You can also configure the quota project by setting the
+    /// `GOOGLE_CLOUD_QUOTA_PROJECT` environment variable. The environment
+    /// variable takes precedence over this option's value.
+    ///
+    /// [quota project]: https://cloud.google.com/docs/quotas/quota-project
+    pub fn set_quota_project<T: Into<String>>(mut self, v: T) -> Self {
+        self.quota_project = Some(v.into());
+        self
+    }
+}
+
+/// Create credentials that authenticate using an [API key].
+///
+/// API keys are convenient because no [principal] is needed. The API key
+/// associates the request with a Google Cloud project for billing and quota
+/// purposes.
+///
+/// Note that most Cloud APIs do not support API keys, instead requiring full
+/// credentials.
```

Comment:
This comment may become stale. Can we say "some Cloud APIs"?  Can we reference a list of APIs that support them?